File: go/election-winners/election-winners.go
package main

import(
	"fmt"
	"slices"
)

func electionWinners(votes []int, k int) int{
	maxVotesBefore := slices.Max(votes)
	possibleWinners := 0

	if k == 0{
		for _, vote := range votes{
			if vote == maxVotesBefore{
				possibleWinners++
			}
		}
		if possibleWinners == 1{
			return 1
		}else{
			return 0
		}
	}

	for _, vote := range votes{
		if maxVotesBefore < vote + k{
			possibleWinners++
		}
	}
	return possibleWinners
}

func main(){
	votes := []int{2, 3, 5, 2}
	k := 3

	fmt.Printf("With the votes of each candidate represented by %v and a remaining of %d vote/s, there could be %d winner/s", votes, k, electionWinners(votes, k) )
}

File: js/election-winners.js
function electionWinners(votes, k){
    let maxVotesBefore = Math.max(...votes)
    let possibleWinners = 0

    
    if(k==0){
        for(vote of votes){
            if(vote == maxVotesBefore)
                possibleWinners++

        }
        return possibleWinners == 1 ? 1 : 0
    }
    
    for(vote of votes){
        if(maxVotesBefore < vote + k){
            possibleWinners++
        }
    }
    
    return possibleWinners
}

let votes = [2, 3, 5, 2], k =3

console.log(`With the votes of each candidate represented by [${votes}] and a remaining of ${k} vote/s, there could be`,electionWinners(votes, k) ,`winner/s`)